controller.go

```diff
@@ -63,17 +72,18 @@ func (c *K8SController) mainloop(item interface{}) {
 		c.OnUpdate(obj)
 	case ControllerEventTypeDelete:
 		c.OnDelete(obj)
+	default:
+		log.Warn().Msgf("no event handler for '%s', event type '%s'", event.Key, event.Type)
 	}
-	c.queue.Done(item)
 }
 
-// Start - starts the informer faktory and sync's the data.
+// Start - starts the informer factory and syncs the data.
 // The wait group passed in is used to track when the informer has gone
-// through 1 full loop and syncronized all the k8s data 1 time
+// through 1 full loop and syncronized all the k8s data exactly 1 time
 func (c *K8SController) Start(wg *sync.WaitGroup) {
 	defer runtime.HandleCrash()
-	defer c.queue.ShutDown()
 	if wg != nil {
+		defer c.queue.ShutDown()
```

Comment:
I know we are going to refactor this in https://github.com/OpsLevel/team-platform/issues/184 but in the mean time we need to move the defer's to inside the goroutine below to not break things.

Reply:
@rocktavious pushed a commit now to defer crash handling inside the goroutine. 

The call to `defer c.queue.ShutDown()` has to stay where it is now since that call is what is currently being used to stop the goroutine.

Reply:
Ok